packages/vite/src/client/client.ts

```diff
@@ -15,7 +15,7 @@ console.log('[vite] connecting...')
 // use server configuration, then fallback to inference
 const socketProtocol =
   __HMR_PROTOCOL__ || (location.protocol === 'https:' ? 'wss' : 'ws')
-const socketHost = `${__HMR_HOSTNAME__ || location.hostname}:${__HMR_PORT__}`
+const socketHost = `${__HMR_HOSTNAME__ || location.hostname}:${__HMR_PORT__ || location.port}`
```

Comment:
Looks like `__HMR_PORT__` will never be falsy. Need to change this as well
https://github.com/vitejs/vite/blob/059070e10aaee27a03ec5fd52e77fa04effe4c8f/packages/vite/src/node/plugins/clientInjections.ts#L50-L50